content/tutorials/spatial_point_in_polygon/index.Rmd

```diff
@@ -0,0 +1,188 @@
+---
+title: "How to make spatial joins (point in polygon)?"
+description: "Various ways to make a spatial join between GIS point data and polygon data"
+author: "Stijn Van Hoey, Hans Van Calster"
+date: 2019-06-20
+categories: ["r", "gis"]
+tags: ["gis", "r", "maps"]
+output: 
+    md_document:
+        preserve_yaml: true
+        variant: markdown_github
+---
+
+```{r include = FALSE}
+library(knitr)
+opts_chunk$set(
+  echo = TRUE
+)
+```
+
+
+```{r}
+library(R.utils)
+library(rgdal)
+library(tidyverse)
+library(leaflet)
+library(sp)
+library(sf)
+library(rgbif)
+library(DBI)
+```
+
+## What we want to do
+
+In this short tutorial, we explore various options to deal with the situation where we have (1) a spatially referenced GIS file with polygons and (2) a spatially referenced set of points that overlaps with the GIS polygons. 
+
+Typically, both data sources contain information (apart from the spatial locations) that needs to be related to each other in some way. In this case study, we want to know for each point in which polygon it is located.
+
+## Get some data to work with
+
+For the point data, we will work with data on the invasive species - Chinese mitten crab (**Eriocheir sinensis**) in  Flanders, Belgium, from the year 2008 (GBIF.org (20th June 2019) GBIF Occurrence Download https://doi.org/10.15468/dl.decefb).
+
+We will use to convenience functions from the `rgbif` package to download the data as a zip file and to import the data as a `tibble` in the R environment. 
+
+```{r}
+invasive_species <- occ_download_get("0032582-190415153152247",
+                                     path = tempdir(),
+                                     overwrite = TRUE) %>% 
+  occ_download_import() %>%
+  filter(!is.na(decimalLongitude), !is.na(decimalLatitude))
+```
+
+We will use the European reference grid system from the European Environmental Agency as an example of a GIS vector layer (each grid cell is a polygon). The Belgium part of the grid system can be downloaded as a sqlite/spatialite database from the EEA website using the following code:
+
+```{r}
+# explicitly set mode = "wb", otherwise zip file will be corrupt
+download.file("https://www.eea.europa.eu/data-and-maps/data/eea-reference-grids-2/gis-files/belgium-spatialite/at_download/file", destfile = paste0(tempdir(), "\\Belgium_spatialite.zip"), mode = "wb")
+
+# this will extract a file Belgium.sqlite to the temporary folder
+unzip(zipfile = paste0(tempdir(), "\\Belgium_spatialite.zip"), exdir = tempdir())  
+```
+
+Next, we use `rgdal::readOGR` to connect with the sqlite/spatialite database and extract the Belgium 10 km x 10 km grid as a SpatialPolygonsDataFrame. Apart from the 10 km x 10 km grid, the database also contains 1 km x 1 km and 100 km x 100 km grids as raster or vector files.
+
+```{r}
+be10grid <- readOGR(dsn = paste0(tempdir(), "\\Belgium.sqlite"), 
+                  layer = "be_10km")
+```
+
+
+
+## Point in polygon with the `sp` package
```

Comment:
Considering the preference for the `sf` package these days (tidyverse compatible etc.), I'd suggest to move this part to the end of the tutorial. Or even, split off into a separate sp-flavoured tutorial.

Reply:
I agree. Will change this, but keeping the sp-way in the same tutorial.